unstructured/ingest/connector/jira.py

```diff
@@ -0,0 +1,491 @@
+            jira = self.session_handle.service
+            issue = jira.issue(self.file_meta.issue_key)
+            issue_fields = nested_object_to_field_getter(issue["fields"])
+
+            self.metadata_fields = {
+                "date_modified": str(issue_fields["updated"]),
+                "date_created": str(issue_fields["created"]),
+                "date_processed": str(datetime.datetime.now().time()),
+                "record_locator": self.record_locator,
+            }
+
+            self.check_exists = True
+            self.file_exists = True
+
+        except HTTPError as error:
+            if error.response.reason == "Not Found":
+                self.check_exists = True
+                self.file_exists = False
+            else:
+                logger.error(f"Error: {error} for issue {self.file_meta.issue_key}")
+
+    @property
+    def date_created(self) -> Optional[str]:
+        if not hasattr(self, "metadata_fields"):
+            self.get_metadata_fields()
+
+        return self.metadata_fields["date_created"] if hasattr(self, "metadata_fields") else None
+
+    @property
+    def date_modified(self) -> Optional[str]:
+        if not hasattr(self, "metadata_fields"):
+            self.get_metadata_fields()
+
+        return self.metadata_fields["date_modified"] if hasattr(self, "metadata_fields") else None
+
+    @property
+    def date_processed(self) -> Optional[str]:
+        if not hasattr(self, "metadata_fields"):
+            self.get_metadata_fields()
+
+        return self.metadata_fields["date_processed"] if hasattr(self, "metadata_fields") else None
+
+    @property
+    def exists(self) -> Optional[bool]:
+        """Whether the document exists on the remote source."""
+        if self.check_exists:
+            return self.file_exists
+
+        self.get_metadata_fields()
+
+        return self.file_exists
+
+    def grouping_folder_name(self):
+        if self.file_meta.board_id:
+            return self.file_meta.board_id
+        else:
+            return self.file_meta.project_id
+
+    @property
+    def filename(self):
+        download_file = f"{self.file_meta.issue_id}.txt"
+
+        return (
+            Path(self.standard_config.download_dir) / self.grouping_folder_name() / download_file
+        ).resolve()
+
+    @property
+    def _output_filename(self):
+        """Create output file path."""
+        output_file = f"{self.file_meta.issue_id}.json"
+
+        return (
+            Path(self.standard_config.output_dir) / self.grouping_folder_name() / output_file
+        ).resolve()
+
+    @requires_dependencies(["atlassian"], extras="jira")
+    @BaseIngestDoc.skip_if_file_exists
+    def get_file(self):
+        logger.debug(f"Fetching {self} - PID: {os.getpid()}")
+
+        try:
+            # GET issue data
+            jira = self.session_handle.service
+            issue = jira.issue(self.file_meta.issue_key)
+            self.check_exists = True
+            self.file_exists = True
```

Comment:
let's make the file_exists operate entirely independently of get_file. i.e. scrap the check_exists etc and just depend on a call to fetch the issue. you can always use cached_property so we only do that once. could also potentially make the the issue a cached property I believe?

Reply:
bump ^

Reply:
You mean ditching the cached property and intentionally making another request right?

If yes, addressed with https://github.com/Unstructured-IO/unstructured/pull/1238/commits/0d50ec28d8eca0a6c071d82998015251a18c804b

If no, please let me know